### PR TITLE
Incorporating content and sections illustrating the information model in English text

### PR DESCRIPTION
**Pull request by henkbirkholz**

### CODE REVIEW

**Comment by henkbirkholz** [comment]
The content of this branch is flagged as a draft as it includes proposals that we will have to agree on (or not) and then massage the content to reflect the consensus.

**Comment by thomas-fossati** [comment]
LOLz for the branch name :-)

File: draft-birkholz-rats-corim.md [hunk 1]
```diff
@@ -100,9 +100,31 @@ While each specification defines its own start rule, only CoMID and CoSWID are s
...
+
+Both types of extensibility also allow for the definition of new nested maps that again in includes additionally defined keys.
+
+## Type Names, Prefixes, and Namespaces
+
+The semantics of the information elements (attributes) defined by CoRIM, CoMID, and CoSWID are sometimes very similar, but do not share the same scope or are slightly different. In order to not overload the semantics of the already existing semantics of the software-centric IANA registries of CoSWID with, for example, hardware-centric semantics of CoMID, new type names are introduce. For example: both CoSWID and CoMID define a tag-id. As CoSWID already specifies "tag-id", CoMID prefixes that type name with "comid." to disambiguate the context, resulting in "comid.tag-id". This prefixing provides a well-defined scope for the use of that the named type defined in this document and guarantees interoperability (no type name collisions) with the CoSWID CDDL definition. Effectively, the prefixes used in this specification enable simple hierarchical namespaces. The prefixing introduced is also based on the anticipated namespace features for CDDL. <!-- FIXME: ref to upcoming CDDL Namespaces I-D -->
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:125]:
```suggestion
The semantics of the information elements (attributes) defined by CoRIM, CoMID, and CoSWID are sometimes very similar, but do not share the same scope or are slightly different. In order to not overload the already existing semantics of the software-centric IANA registries of CoSWID with, for example, hardware-centric semantics of CoMID, new type names are introduced. For example: both CoSWID and CoMID define a tag-id. As CoSWID already specifies `tag-id`, CoMID prefixes that type name with `comid.` to disambiguate the context, resulting in `comid.tag-id`. This prefixing provides a well-defined scope for the use of the types defined in this document and guarantees interoperability (no type name collisions) with the CoSWID CDDL definition. Effectively, the prefixes used in this specification enable simple hierarchical namespaces. The prefixing introduced is also based on the anticipated namespace features for CDDL. <!-- FIXME: ref to upcoming CDDL Namespaces I-D -->
```

**Comment by thomas-fossati** [comment]
Thanks Henk for the herculean effort!  To me this looks good enough for a -00, and I'd be happy to publish as-is.  But I'll let @yogeshbdeshpande and @nedmsmith chime in in case they have fixes to suggest. In any case, if there are bits to improve / add, which I'm sure there are, I reckon we could use the coming two weeks to do a further pass over the document and publish a -01 as soon as the window reopens just before the meeting starts.

File: draft-birkholz-rats-corim.md [hunk 2]
```diff
@@ -132,17 +353,436 @@ concise-mid-tag = {
...
+: A directed property that associates key material used to identify a module instance or a module class that is an identifying part of a device(-set).
+
+$$triples-map-extension:
+
+: This CDDL socket is used to add new information elements to the triples-map container. See FIXME.
+
+{: #model-environment-map}
+## The environment-map Container
+
+This map represents the module(s) that a triple-map can point directed properties (relationships) from in order to associate them with external information for remote attestation, such as reference values, endorsement and endorsed values, verification key material for evidence, or identifying key material for module (re-)identification. This map can identify a single module instance via `comid.instance` or groups of modules via `comid.group`. Referencing classes of modules requires the use of the more complex `class-map` container. This rule and its constraints MUST be followed when generating or validating a CoMID tag.
```
Comment (yogeshbdeshpande) [inline draft-birkholz-rats-corim.md:518]: The document and text looks great! Few minor comments here!

1) Kindly introduce class concept to the reader first and then instance and group concept.

2) Perhaps the class definition can be slightly simplified, on the following lines:
This map can identify a either a module class which identifies a device family  via `comid.class`  or a single module instance via via `comid.instance` or groups of modules via `comid.group`.

File: draft-birkholz-rats-corim.md [hunk 3]
```diff
@@ -132,17 +353,436 @@ concise-mid-tag = {
...
+$class-id-type-choice /= tagged-uuid-type
+~~~~
+
+The following describes each member of the class-map container.
+
+{: #model-measurement-values-map}
+## The measurement-map and measurement-values-map Containers
+
+One of the targets (range) that a triple-map can point to in order to associate it with a module (domain) is the measurement-map. This map is used to provide reference measurements values that can be compared with Evidence Claim values or Endorsements and endorsed values from other sources than the corresponding CoRIM. `measurement-map` comes with a measurement key that identifies the measured element with via a OID reference or a UUID. `measurement-values-map` contains the actual measurements associated with the module(s). Byte strings with corresponding bit masks that highlights which bits in the byte string are used as reference measurements or endorsement are located in `raw-value-group`. The members of `measurement-values-map` provide well-defined and well-scoped semantics for reference measurement or endorsements with respect to a given module instance, class, or group. This rule and its constraints MUST be followed when generating or validating a CoMID tag.
```
Comment (yogeshbdeshpande) [inline draft-birkholz-rats-corim.md:571]: Considering the length of statement, request please consider simple split on the following lines:

This map is used to provide (a) reference measurements values that can be compared with Evidence Claim values - reported from other sources, example attesting device.
AND/OR (b) Endorsements that can be compared with endorsed values received from other sources than the corresponding CoRIM.

**Pull request closed without merging**